addons/escoria-core/plugin.gd

```diff
@@ -3,16 +3,16 @@ tool
 extends EditorPlugin
 
 
-# Setup Escoria
-func _enter_tree():
+# Virtual function called when plugin is enabled.
+func _enable_plugin():
 	add_autoload_singleton(
 		"escoria",
 		"res://addons/escoria-core/game/escoria.tscn"
 	)
 
 	# Add input actions in InputMap
-	InputMap.add_action(escoria.inputs_manager.SWITCH_ACTION_VERB)
-	InputMap.add_action(escoria.inputs_manager.ESC_SHOW_DEBUG_PROMPT)
+	InputMap.add_action(get_node("escoria").inputs_manager.SWITCH_ACTION_VERB)
```

Comment:
Suggestion : You might want to even create a wrapper for get_node("escoria") that acts as some sort of factory. Something like Escoria.get_escoria(). 

If the node is not present, display a message in the console that says to the user that Escoria core is missing (or disabled) and that it's not the intended behaviour. And what they should do to fix it (re-enable plugin or whatnot).

Reply:
@BHSDuncan but if you can't refer to the singleton instance by a hard-coded name ("escoria") then how can you ever refer to it? By getting the (hopefully) only root node's child that has the Escoria class name?

Reply:
What are you talking about? My point is that using that string EVERYWHERE is an issue, and to reduce it down to a single location. I'm effectively agreeing with what you're suggesting.